python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -380,3 +381,89 @@ def close(self) -> None:
         if self.is_local_node(self._reader_node_id):
             self.ensure_registered_as_reader()
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
+
+
+@PublicAPI(stability="alpha")
+class MultiChannel(ChannelInterface):
+    def __init__(
+        self,
+        writer: Optional[ray.actor.ActorHandle],
+        readers: List[Optional[ray.actor.ActorHandle]],
+        _local_channel: Optional[LocalChannel] = None,
+        _remote_channel: Optional[Channel] = None,
+    ):
+        """
+        Can be used to send data to different readers via different channels.
+        For example, if the reader is in the same worker process as the writer,
+        the data can be sent via LocalChannel. If the reader is in a different
+        worker process, the data can be sent via shared memory channel.
+        """
+        self._writer = writer
+        self._readers = readers
+        self._local_channel = _local_channel
+        self._remote_channel = _remote_channel
+
+        remote_readers = []
+        for reader in self._readers:
+            if self._writer != reader:
+                remote_readers.append(reader)
+        # There are some local readers which are the same Ray actor as the writer.
```

Comment:
Can you please clarify that they are in the same address space?

Reply:
I updated the wording a bit and explicitly mentioned that they are in the "same worker process". For details, see https://github.com/ray-project/ray/commit/d1a025c050cb80500d76da8f53ff49f0fc2fe0f1. Is this what you expected?